src/Runtime/include/OnnxMlir.h

```diff
@@ -0,0 +1,289 @@
+ *
+ * Set the data sizes array of the RtMemRef to the values in the input array.
+ */
+void rmrSetDataShape(RtMemRef *rmr, INDEX_TYPE *dataSizes);
+
+/**
+ * RtMemRef data strides getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return pointer to the data strides array.
+ */
+int64_t *rmrGetDataStrides(RtMemRef *rmr);
+
+/**
+ * RtMemRef data strides setter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @param dataStrides, data strides array to be set
+ *
+ * Set the data strides array of the RtMemRef to the values in the input array.
+ */
+void rmrSetDataStrides(RtMemRef *rmr, int64_t *dataStrides);
+
+/**
+ * RtMemRef data type getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return ONNX data type of the data buffer elements.
+ */
+int rmrGetDataType(RtMemRef *rmr);
+
+/**
+ * RtMemRef data type setter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @param dataType, ONNX data type to be set
+ *
+ * Set the ONNX data type of the data buffer elements.
+ */
+void rmrSetDataType(RtMemRef *rmr, int dataType);
+
+/**
+ * RtMemRef data buffer size getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return the total size of the data buffer in bytes.
+ */
+// int64_t rmrGetDataBufferSize(RtMemRef *rmr);
+
+/**
+ * RtMemRef rank getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return rank of data sizes and strides of the RtMemRef.
+ */
+int rmrGetRank(RtMemRef *rmr);
+
+/**
+ * RtMemRef name getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return pointer to the name of the RtMemRef,
+ *         an empty string if the name is not set.
+ */
+char *rmrGetName(RtMemRef *rmr);
+
+/**
+ * RtMemRef name setter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @param name, name of the RtMemRef to be set
+ *
+ * Set the name of the RtMemRef.
+ */
+void rmrSetName(RtMemRef *rmr, char *name);
+
+/**
+ * RtMemRef number of elements getter
+ *
+ * @param rmr, pointer to the RtMemRef
+ * @return the number of elements in the data buffer.
+ */
+INDEX_TYPE rmrGetNumElems(RtMemRef *rmr);
+
+/*---------------------------------------- */
+/* C/C++ API for RtMemRefList calls */
+/*---------------------------------------- */
+
+/**
+ * RtMemRefList creator
+ *
+ * @param rmrs, array of pointers to RtMemRef
+ * @param n, number of elements in rmrs array
+ * @return pointer to the RtMemRefList created, NULL if creation failed.
+ *
+ * Create an RtMemRefList with specified RtMemRef array.
+ * If a RtMemRef has a name, in addition to be accessed by its index,
+ * the RtMemRef can also be accessed by its name.
+ */
+RtMemRefList *rmrListCreate(RtMemRef **rmrs, int n);
+
+/**
+ * RtMemRefList destroyer
+ *
+ * @param ormrd, pointer to the RtMemRefList to be destroyed
+ *
+ * Destroy the RtMemRefList struct.
+ */
+void rmrListDestroy(RtMemRefList *ormrd);
+
+/**
+ * RtMemRefList RtMemRef array getter
+ *
+ * @param ormrd, pointer to the RtMemRefList
+ * @return pointer to the array of RtMemRef pointers.
+ */
```

Comment:
This is confusing... why return the list of pointers? I would have preferred to have a 

rmrGetFromList(RtMemRefList *list, uint index)
and rmrGetFromListWithName(RtMemRefList *list, char *name)

Reply:
Now there're both options. More specifically, we use to have index-based mutator and accessor for private use. But I dropped the index-based mutator (because I can think of no legitimate use) and made the accessor public.